mcgill-en.csl

```diff
@@ -275,18 +342,34 @@ all the other tests should have been done... -->
     </group>
   </macro>
   <macro name="render-legislation">
-    <group delimiter=", ">
-      <!-- if no volume, assume bill -->
-      <group delimiter=" ">
-        <text variable="title" font-style="italic"/>
-        <text variable="references" prefix="(" suffix=")"/>
-      </group>
-      <group delimiter=" ">
-        <text variable="container-title"/>
-        <date form="text" variable="issued" date-parts="year"/>
-      </group>
-      <text variable="section"/>
-    </group>
+	  
+	    <choose>
+		  <if variable="references" match="any">
```

Comment:
something weird is going on here -- you're checking for `references` but then not printing them but then you are calling them in the `else` where they can't appear

Reply:
Hi, this is because of a quirk of Canadian Law being both common law and civil law.  
Normally a statute looks  like "title year, section" (Criminal Code, RSC 1985, C 8, art 2).  But in civil law, "Codes" are structured completely differently: "section title (year)" (Art 2 CCQ (1985).)  
We need both citation styles, but I only have the one type.  It requires an if-then, and I opted to use references.  It can only check for the presence of a value and cannot operate on the values themselves.  I didn't have a lot of fields to work with and references was available.
When there is a value in references, it formats as a Code; otherwise it is as an english statute.
I hope this explains it.

However, I will remove the references variable from the else statement, since it must be null there.

S